arch/xtensa/src/esp32/esp32_spiram.c

```diff
@@ -238,6 +245,8 @@ void IRAM_ATTR esp_spiram_init_cache(void)
   /* Flush and enable icache for APP CPU */
 
 #ifdef CONFIG_SMP
+  cache_flush(cpu);
+  cache_read_enable(cpu);
```

Comment:
```suggestion
  cache_flush(APP_CPU_NUM);
  cache_read_enable(APP_CPU_NUM);
```

Reply:
oops, it was relying on other local changes. fixed. thank you.